extensions/amp-autocomplete/0.1/amp-autocomplete.js

```diff
@@ -14,51 +14,374 @@
+    this.inputElement_.addEventListener('blur', this.hideResults.bind(this));
+    this.renderResults_();
     return Promise.resolve();
   }
 
+  /**
+   * Create and return <div> element from given plan-text item.
+   * @param {string} item
+   * @return {!HTMLElement}
+   * @private
+   */
+  createElementFromItem_(item) {
+    const element = this.element.ownerDocument.createElement('div');
+    element.classList.add('i-amphtml-autocomplete-item');
+    element.setAttribute('role', 'listitem');
+    element.addEventListener('mousedown', this.mousedownHandler_.bind(this));
+    element.textContent = item;
+    return element;
+  }
+
+  /**
+  * Handle rendering results on user input.
+  * @param {?event} event
+  * @private
+  */
+  inputHandler_(event) {
+    if (event.inputType === 'deleteContentBackward') {
+      // Explore options for caching results to avoid repetitive queries.
+    }
+    this.renderResults_();
+    this.showResults();
+  }
+
+  /**
+   * Render filtered results on the current input and update the container_.
+   * @private
+   */
+  renderResults_() {
+    const userInput = this.inputElement_.value;
+    this.clearAllItems();
+    if (userInput.length < this.minChars_) {
+      return;
+    }
+    const filteredData = this.filterData_(this.inlineData_, userInput);
+    filteredData.forEach(item => {
+      this.container_.appendChild(this.createElementFromItem_(item));
+    });
+  }
+
+  /**
+   * Apply the filter to the given data based on the given input.
+   * @param {!Array<string>} data
+   * @param {string} input
+   * @return {!Array<string>}
+   * @private
+   */
+  filterData_(data, input) {
+    let filteredData = data.filter(item => {
+      switch (this.filter_) {
+        case FilterType.SUBSTRING:
+          return item.includes(input);
+          break;
+        case FilterType.PREFIX:
+          return item.startsWith(input);
+          break;
+        case FilterType.TOKEN_PREFIX:
+          return item.split(' ').some(token => {
+            return token.startsWith(input);
+          });
+          break;
+        case FilterType.FUZZY:
+          throw new Error(`Filter not yet supported: ${this.filter_}`);
+          break;
+        case FilterType.CUSTOM:
+          throw new Error(`Filter not yet supported: ${this.filter_}`);
+          break;
+        case FilterType.NONE:
+          // Query server endpoint.
+          throw new Error(`Filter not yet supported: ${this.filter_}`);
+          break;
+        default:
+          throw new Error(`Unexpected filter: ${this.filter_}`);
+      }
+    });
+
+    // Truncate to max-entries.
+    if (this.maxEntries_ && this.maxEntries_ < filteredData.length) {
+      filteredData = filteredData.slice(0, this.maxEntries_);
+    }
+
+    return filteredData;
+  }
+
+  /** Set container_ visibility to visible. */
+  showResults() {
+    setStyle(this.container_, 'visibility', 'visible');
```

Comment:
I think the `visibility` property might not be the best mechanism to use to hide the results list. I would first recommend using `src/style.js#toggle`

https://github.com/ampproject/amphtml/blob/e194f8e0726b8a3241dcac2001c43669a3df88ae/src/style.js#L214-L219